pkg/apiserver/internal/v1/account.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2020 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/catalog/v1alpha1"
+	v1 "github.com/kubermatic/kubecarrier/pkg/apiserver/api/v1"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/oidc"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/util"
+)
+
+type accountServer struct {
+	client client.Client
+}
+
+var _ v1.AccountServiceServer = (*accountServer)(nil)
+
+// +kubebuilder:rbac:groups=catalog.kubecarrier.io,resources=accounts,verbs=get;list;watch
+
+func NewAccountServiceServer(c client.Client) v1.AccountServiceServer {
+	return &accountServer{
+		client: c,
+	}
+}
+
+func (o accountServer) List(ctx context.Context, req *v1.AccountListRequest) (res *v1.AccountList, err error) {
+	user, present := oidc.ExtractUserInfo(ctx)
+	if !present {
+		return nil, status.Error(codes.Unauthenticated, "unauthenticated user")
+	}
+	return o.handleListRequest(ctx, req, user.GetName())
+}
+
+func (o accountServer) handleListRequest(ctx context.Context, req *v1.AccountListRequest, username string) (res *v1.AccountList, err error) {
+	var listOptions []client.ListOption
+	listOptions, err = o.validateListRequest(req)
+	listOptions = append(listOptions, accountByUsernameListOption(username))
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	accountList := &catalogv1alpha1.AccountList{}
```

Comment:
In the future, we should filter the account list depending on which accounts the users has meaningful permissions towards.

Reply:
What do you mean by users have meaningful permissions towards?
Currently, we are listing account which has the user in its `.Spec.Subjects` which means we are listing account which user has access to.

Reply:
oh ok; I haven't noticed you're doing filtering via the field index; this is fine.